versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/UseNextSnapshotsMojo.java

```diff
@@ -98,76 +80,43 @@ public UseNextSnapshotsMojo(
         allowSnapshots = true;
     }
 
-    /**
-     * @param pom the pom to update.
-     * @throws org.apache.maven.plugin.MojoExecutionException when things go wrong
-     * @throws org.apache.maven.plugin.MojoFailureException   when things go wrong in a very bad way
-     * @throws javax.xml.stream.XMLStreamException            when things go wrong with XML streaming
-     * @see org.codehaus.mojo.versions.AbstractVersionsUpdaterMojo#update(MutableXMLStreamReader)
-     */
-    protected void update(MutableXMLStreamReader pom)
-            throws MojoExecutionException, MojoFailureException, XMLStreamException, VersionRetrievalException {
-        try {
-            DependencyManagement dependencyManagement =
-                    PomHelper.getRawModel(getProject()).getDependencyManagement();
-            if (dependencyManagement != null) {
-                useNextSnapshots(
-                        pom,
-                        dependencyManagement.getDependencies(),
-                        DependencyChangeRecord.ChangeKind.DEPENDENCY_MANAGEMENT);
-            }
-            if (getProject().getDependencies() != null && isProcessingDependencies()) {
-                useNextSnapshots(pom, getProject().getDependencies(), DependencyChangeRecord.ChangeKind.DEPENDENCY);
-            }
-            if (getProject().getParent() != null && isProcessingParent()) {
-                useNextSnapshots(pom, singletonList(getParentDependency()), DependencyChangeRecord.ChangeKind.PARENT);
-            }
-        } catch (IOException e) {
-            throw new MojoExecutionException(e.getMessage(), e);
-        }
+    @Override
+    protected boolean isAllowMajorUpdates() {
+        return allowMajorUpdates;
+    }
+
+    @Override
+    protected boolean isAllowMinorUpdates() {
+        return allowMinorUpdates;
+    }
+
+    @Override
+    protected boolean isAllowIncrementalUpdates() {
+        return allowIncrementalUpdates;
+    }
+
+    @Override
+    protected boolean isAllowSnapshots() {
+        return true;
```

Comment:
why such ...

Reply:
This is a callback for [UseLatestVersionsMojoBase#update](https://github.com/mojohaus/versions/pull/1202/files#diff-67c5e18947ef94a25779821e3341fd0f5660aa4031c8c1492b252c09cf0f0716R153) so that it selects snapshots -- we are looking for snapshots to upgrade to.

Reply:
The property `allowSnapshots` doesn't make sense in some of the goals, like this one, so I'll push the property down the class hierarchy to goals where it does make sense. At the same time, I'll pull `isAllowSnapshots()` upwards as an abstract method.

Reply:
Please note that I'm doing this in stages so that the PR doesn't grow too big.

Reply:
test comment